core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletMetadata.java

```diff
@@ -435,4 +445,34 @@ static TabletMetadata create(String id, String prevEndRow, String endRow) {
     te.fetchedCols = EnumSet.of(ColumnType.PREV_ROW);
     return te;
   }
+
+  public static synchronized Set<TServerInstance> getLiveTServers(ClientContext context) {
+    final Set<TServerInstance> currentServers = new HashSet<>();
+
+    final String path = context.getZooKeeperRoot() + Constants.ZTSERVERS;
+
+    HashSet<String> all = new HashSet<>(context.getZooCache().getChildren(path));
+
+    for (String zPath : all) {
+      checkServer(context, currentServers, path, zPath);
+    }
```

Comment:
This pattern of populating `currentServers` inside the other method, `checkServer`, after looping over `all` can be improved with something like:

```java
  var children = context.getZooCache().getChildren(path);
  var liveTservers = children.map(TabletMetadata::checkServer).filter(Optional::isPresent).collect(Collectors.toSet());
  log(liveTservers);
  return liveTservers;
```

This requires `checkServer` to be modified to return either `Optional.of(instance)` or `Optional.empty()`, but I think it's more readable.

Reply:
I disagree.  To me what you have is longer and less readable.  I will rename all to children though.

Reply:
The main point I find difficult to read is the pass by reference of the data structure to a different method that is responsible for updating the data structure, conditionally. That kind of pattern can be hard to follow.

So, instead of:
```java
  methodToUpdateDataStructure(x1, x2, x3, dataStructure)
```

You can do:
```java
  var obj = shouldAddToDataStructure(x1, x2, x3);
  if (obj != null)
    dataStructure.add(obj);
```

So, if the part of my suggestion that was less readable to you was the stream stuff, here's a non-stream version:

```suggestion
    var currentServers = new HashSet<TServerInstance>();
    for (var child : context.getZooCache().getChildren(context.getZooKeeperRoot() + Constants.ZTSERVERS)) {
      var instance = checkServer(child);
      if (instance != null)
        currentServers.add(instance);
    }
```

I just thought the stream version was better, but honestly, anything is better than the pass-by-reference mutable data structure pattern, IMO.

If you prefer the current implementation, that's fine too. I don't feel very strongly about it. It was just something I noticed and thought could be improved.

Reply:
I agree that mutating a data structure passed as a param can be deceptive.  For me personally, I typically find streams to be less readable.  I was just using what was there to start with since some of the code used in the locking I don't fully understand.  I do like your non-stream suggestion, I will use that.

Reply:
@ctubbsii I think I came up with a nice combo of ideas in ba8fb27